Assets/Scripts/NPC/Grid.cs

```diff
@@ -1,9 +1,30 @@
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.Serialization;
 
 public sealed class Grid : MonoBehaviour
 {
     [SerializeField] private List<Waypoint> waypoints = new List<Waypoint>();
+    
+    [Header("Grid Creation")]
+    [SerializeField] private Waypoint waypointPrefab;
+    [SerializeField] private Vector2 gridSize;
+    [SerializeField] private bool createGrid;
+    
+    private Vector2 initPosition;
+    private Vector2 safeZone;
+    private Vector2 waypointAmount;
+    private bool hasUpdatedGrid;
+    private int waypointIndex;
+
+    private void Awake()
+    {
+        createGrid = false;
+        if (waypoints.Count <= 0)
+        {
+            CreateGrid();
+        }
```

Comment:
```suggestion
        if (waypoints.Count <= 0) CreateGrid();
```
Deze if statement is kort genoeg om op 1 line te zetten naar mijn mening.